Modules/_ctypes/libffi_osx/ffi.c

```diff
@@ -161,8 +161,8 @@ ffi_prep_cif(
 	/* Perform a sanity check on the return type */
 	FFI_ASSERT_VALID_TYPE(cif->rtype);
 
-	/* x86-64 and s390 stack space allocation is handled in prep_machdep.  */
-#if !defined M68K && !defined __x86_64__ && !defined S390 && !defined PA
+	/* x86-64 and s390x stack space allocation is handled in prep_machdep.  */
+#if !defined M68K && !defined __x86_64__ && !defined PA
```

Comment:
This is in `libffi_osx` directory, so unrelated to support for s390 on Linux.
To verify correctness of changes in this file, somebody would have to provide relevant output from compiler on Mac OS X on s390 and s390x.
E.g. gcc on Linux on s390**x** defines both `__s390__` and `__s390x__` constants ([gcc-s390x.txt](https://bugs.python.org/file49818/gcc-s390x.txt)).

Reply:
I don't understand this comment. Does macOS / Darwin even compile on s390? "Mach" here properly refers to the Mach microkernel, not to macOS support.